de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/celleditor/RequirementsAttributeValuePerColumnEditingSupport.java

```diff
@@ -0,0 +1,122 @@
+
+	protected final int attributeIndex;
+
+	protected final VirSatTransactionalEditingDomain domain;
+
+	/**
+	 * @param editingDomain
+	 *            the editing domain
+	 * @param viewer
+	 *            the column viewer
+	 * @param attIndex
+	 *            the index of the attribute definition this class should support
+	 */
+	public RequirementsAttributeValuePerColumnEditingSupport(EditingDomain editingDomain, ColumnViewer viewer, int attIndex) {
+		super(editingDomain, viewer, getAttributeValueProperty((VirSatTransactionalEditingDomain) editingDomain));
+		this.attributeIndex = attIndex;
+		this.domain = (VirSatTransactionalEditingDomain) editingDomain;
+	}
+
+
+	@Override
+	protected APropertyInstance getPropertyInstance(Object element) {
+		ComposedPropertyInstance cpi = (ComposedPropertyInstance) element;
+		Requirement requirement = new Requirement(cpi.getTypeInstance());
+
+		if (requirement.getReqType() == null 
+				|| attributeIndex >= requirement.getReqType().getAttributes().size()) {
+			return null;
+		}
+
+		RequirementAttribute attributeDef = requirement.getReqType().getAttributes().get(attributeIndex);
+
+		APropertyInstance instance = null;
+		for (AttributeValue value : requirement.getElements()) {
+			if (value.getAttType().equals(attributeDef)) {
+				instance = getPropertyInstanceFromBean(value);
+			}
+		}
+		if (instance == null) {
+			// Requirement attribute instance does not exist yet, create one...
+			// But don't add it to the model yet, otherwise we will trigger a notification
+			// that disturbs UI (Focus loss)
+			AttributeValue newAttributeInstance = new AttributeValue(getConcept());
+			instance = getPropertyInstance(newAttributeInstance);
+		}
+
+		return instance;
+
+	}
+
+	/**
+	 * get the property instance from the bean object
+	 * 
+	 * @param attributeInstance
+	 *            the bean object
+	 * @return the property instance
+	 */
+	protected APropertyInstance getPropertyInstanceFromBean(AttributeValue attributeInstance) {
+		CategoryAssignmentHelper attributeInstanceHelper = new CategoryAssignmentHelper(
+				attributeInstance.getTypeInstance());
+		return attributeInstanceHelper.getPropertyInstance(AttributeValue.PROPERTY_VALUE);
+	}
+
+	/**
+	 * Get the attribute definition of the current editor element
+	 * 
+	 * @param element
+	 *            the editor element
+	 * @return the attribute definition
+	 */
+	@Override
+	protected RequirementAttribute getAttributeDefinition(Object element) {
+		Requirement requirement = null;
+		
+		if (element instanceof ComposedPropertyInstance) {
+			ComposedPropertyInstance cpi = (ComposedPropertyInstance) element;
+			requirement = new Requirement(cpi.getTypeInstance());
+		} else if (element instanceof Requirement) {
+			requirement = (Requirement) element;
+		} else if (element instanceof CategoryAssignment) {
+			requirement = new Requirement((CategoryAssignment) element);
+		}
+		
+		if (requirement == null) {

```

Comment:
Maybe we can make this an else case in the above if else chain.
Because if the variable `requirement` happens to be null despite going into one of the cases something is wrong.